### DISCUSSION TITLE
Handling filenames

### DISCUSSION THREAD

**Discussion by genox** [tinacms/tinacms#3361] [Help]
While editing a document, is it somehow possible to

1. check if a filename already exists while generating a new filename?
2. rename a file, to reflect changes in fields
3. use the content of a field to display in the collection overview instead of just the filename? 

This would essentially allow to abstract away the filename and remove the need to think about that.

**Comment by jamespohalloran** [comment]
> check if a filename already exists while generating a new filename?

Good catch, just [filed an issue](https://github.com/tinacms/tinacms/issues/3362)

> use the content of a field to display in the collection overview instead of just the filename?

You can accomplish this with the [`isTitle` property](https://tina.io/docs/reference/fields/#istitle) on a field. Is that what you're looking for?
<img width="1138" alt="Screen Shot 2022-11-02 at 10 11 03 AM" src="https://user-images.githubusercontent.com/3323181/199498681-beb10625-313b-410e-b276-cb9a1c1a4aa0.png">
